### DISCUSSION TITLE
Unexpected token 'default' bug

### DISCUSSION THREAD

**Discussion by dangreen** [storybookjs/storybook#13935] [Help]
**Describe the bug**

```
Unexpected token 'default'
...
try {
    // @ts-ignore
    default.preloadState.displayName = "default.preloadState";
    // @ts-ignore
    default.preloadState.__docgenInfo = { "description": "", "displayName": "default.preloadState", "props": {} };
...
```

**To Reproduce**

Example component code:

```
export default function Component() {
  ...
}

function fn() {}

Component.fn = fn
```

all works if change default export:

```
function Component() {
  ...
}

function fn() {}

Component.fn = fn

export default Component
```

**Expected behavior**

No error.

**System**

```
Environment Info:

  System:
    OS: macOS
  Binaries:
    Node: 12.14.1 - ~/.nvm/versions/node/v12.14.1/bin/node
    Yarn: 1.22.4 - ~/.nvm/versions/node/v12.14.1/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.14.1/bin/npm
  npmPackages:
    @storybook/addon-actions: ^6.1.11 => 6.1.11 
    @storybook/addon-backgrounds: ^6.1.11 => 6.1.11 
    @storybook/addon-controls: ^6.1.11 => 6.1.11 
    @storybook/addon-docs: ^6.1.18 => 6.1.18 
    @storybook/addon-knobs: ^6.1.11 => 6.1.11 
    @storybook/addon-links: ^6.1.11 => 6.1.11 
    @storybook/addon-storyshots: ^6.1.11 => 6.1.11 
    @storybook/addon-storyshots-puppeteer: ^6.1.11 => 6.1.11 
    @storybook/addon-toolbars: ^6.1.11 => 6.1.11 
    @storybook/addon-viewport: ^6.1.11 => 6.1.11 
    @storybook/addons: ^6.1.18 => 6.1.18 
    @storybook/react: ^6.1.18 => 6.1.18 
```

**Additional context**
babel-loader plus ts-loader

**Comment by shilman** [comment]
This looks like a babel configuration issue. Do you have a custom babel config? If you remove it does it work?

**Comment by riccjohn** [comment]
I'm having the same issue with my codebase. It looks like if I remove any stories for components that rely on React Context, storybook can run properly. However, even when I add context to the stories, I still get this error.

**Reply by nathanmillar16** [reply]
me too! Did you find a solution?

**Reply by kiastorm** [reply]
> me too! Did you find a solution?

I solved this by separating function declarations from the 'export default' keyword across the codebase.

**Before**
export default function Something() { ... }

**After**
function Something() { ... }

export default Something;

**Reply by htutwaiphyoe** [reply]
This worked for me too. Thanks 🙏
